Use nomad-native attribute for host IP

twunderlich-grapl opened the pull request:
<!-- Thank you for your contribution to Grapl! Please take some time
to fill out this short template to help us review and understand your
PR. -->

### Which issue does this PR correspond to?

<!-- Please provide a link to the GitHub Issue this pull request is
meant to address. -->

### What changes does this PR make to Grapl? Why?
⚠️ This will break anyone's local environment using Nomad prior to 1.3.2. ⚠️  
This switches us to using Nomad's native attribute to get the host IP since Nomad 1.3.2 has been out for over a month

Now that https://github.com/hashicorp/nomad/pull/12817 is merged in as
of Nomad 1.3.2, we should stop using the hax to get the host IP and
instead use Nomad's native method to get the host IP.

<!-- Please describe the functional impact of these changes and the
rationale behind them. This will help us understand your approach to
solving the problem. -->

### How were these changes tested?
Tested locally with make up and checking Jaeger, regression tests may also pick up any issues but I'm not sure if Pulumi/Nomad will pick it up.
<!-- Please describe how these changes were tested. There should be
sufficient detail that reviewers can replicate your testing
procedure. If the procedure is a manual one that's OK, your
description will help us work with you to get it automated. -->

Commit:
Use nomad-native attribute for host IP

Now that https://github.com/hashicorp/nomad/pull/12817 is merged in as
of Nomad 1.3.2, we should stop using the hax to get the host IP and
instead use Nomad's native method to get the host IP.

christophermaier commented on nomad/template_smple.json:
```diff
@@ -0,0 +1,12 @@
+{
```
Was this supposed to have been committed?

twunderlich-grapl replied:
Nope, removing now

Commit:
Merge branch 'main' into twunderlich/switch-to-nomad-native-ip

 Conflicts:
	nomad/grapl-core.nomad
	pulumi/grapl/__main__.py

christophermaier reviewed (approved):
Looks good, with one question.

Commit:
Ugh, vars isn't optional

RM bad code

Commit:
Fix issue when we're sending in empty vars to Nomad

This fixes one issue where the Nomad job doesn't accept any variables
which results in a KeyError, and speeds up cases where we are not
passing in variables
